src/backend/controllers/users.js

```diff
@@ -0,0 +1,168 @@
+import { generateJWT, hashPassword} from '../utils/authorization.js';
+import model from "../model/users.js";
+
+export async function signup(req, res) {
+
+    if (!req.body.StudentID || !req.body.SchoolEmail || !req.body.Username || !req.body.Fname || !req.body.Lname || !req.body.Password) {
+        return res.status(400).send('Missing value');
+    }
+
+    if (!req.body.SchoolEmail.match(/^[a-zA-Z0-9]+@[a-zA-Z0-9]+\.[A-Za-z.]+$/)) {
+        return res.status(400).send('Invalid email format');
+    }
+
+    const user = await model.getUser('SchoolEmail', req.body.SchoolEmail);
+
+    if (user) {
+        return res.status(400).send('Email already exists');
+    }
+
+    const user_id = await model.createUser(req.body.StudentID, req.body.SchoolEmail, req.body.Username, req.body.Fname, req.body.Lname, hashPassword(req.body.Password));
+
+    if (!user_id) {
+        return res.status(500).send('Internal server error');
+    }
+
+    const result = await generateJWT(req.body.StudentID);
+
+    return res.status(200).send({ data: { 'Token': result } });
+
+}
+
+export function signin(req, res) {
+
+    if (!req.body.StudentID || !req.body.Password) {
+        return res.status(400).send('Missing value');
+    }
+
+    model.getUser('StudentID', req.body.StudentID).then((user) => {
+        if (!user) {
+            return res.status(400).send('Email does not exist');
```

Comment:
I think this should return status code 404. Same goes for the "does not exist" errors below. If you do change this, please change the other "does not exist" errors as well.

Reply:
OK